app/src/main/java/com/zeapo/pwdstore/git/GitAsyncTask.kt

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright © 2014-2020 The Android Password Store Authors. All Rights Reserved.
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+package com.zeapo.pwdstore.git
+
+import android.app.Activity
+import android.app.ProgressDialog
+import android.content.Context
+import android.content.Intent
+import android.os.AsyncTask
+import com.github.ajalt.timberkt.e
+import com.zeapo.pwdstore.PasswordStore
+import com.zeapo.pwdstore.R
+import net.schmizz.sshj.userauth.UserAuthException
+import org.eclipse.jgit.api.CommitCommand
+import org.eclipse.jgit.api.GitCommand
+import org.eclipse.jgit.api.PullCommand
+import org.eclipse.jgit.api.PushCommand
+import org.eclipse.jgit.api.RebaseResult
+import org.eclipse.jgit.api.StatusCommand
+import org.eclipse.jgit.transport.RemoteRefUpdate
+import java.io.IOException
+import java.lang.ref.WeakReference
+
+
+class GitAsyncTask(
+    activity: Activity,
+    private val refreshListOnEnd: Boolean,
+    private val operation: GitOperation,
+    private val finishWithResultOnEnd: Intent?) : AsyncTask<GitCommand<*>, Int, GitAsyncTask.Result>() {
+
+    private val activityWeakReference: WeakReference<Activity> = WeakReference(activity)
+    private val activity: Activity?
+        get() = activityWeakReference.get()
+    private val context: Context = activity.applicationContext
+    private val dialog = ProgressDialog(activity)
+
+    sealed class Result {
+        object Ok : Result()
+        data class Err(val err: Exception) : Result()
+    }
+
+    override fun onPreExecute() {
+        dialog.run {
+            setMessage(activity!!.resources.getString(R.string.running_dialog_text))
+            setCancelable(false)
+            show()
+        }
+    }
+
+    override fun doInBackground(vararg commands: GitCommand<*>): Result? {
+        var nbChanges: Int? = null
+        for (command in commands) {
+            try {
+                if (command is StatusCommand) {
```

Comment:
worthwhile making this a `when` clause? IMO it'd be a bit more readable

Reply:
Definitely, done.